C/plugins/north/OMF/linkdata.cpp

```diff
@@ -176,28 +183,33 @@ string OMFLinkedData::processReading(const Reading& reading, const string&  AFHi
 
 			// Create the link for the asset if not already created
 			string link = assetName + "." + dpName;
+			auto dpLookup = m_linkedAssetState->find(link);
+
 			string baseType = getBaseType(dp, format);
-			auto container = m_containerSent->find(link);
-			if (container == m_containerSent->end())
+			if (dpLookup == m_linkedAssetState->end())
+			{
```

Comment:
This should do something even if just logging a fatal error. The following "_if_" on Line 192 should be "_else if_" otherwise the check on 192 will crash if the link lookup has failed.